x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/inference/textembedding/TextEmbeddingOperatorRequestIterator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.inference.textembedding;
+
+import org.elasticsearch.compute.data.BytesRefBlock;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xpack.core.inference.action.InferenceAction;
+import org.elasticsearch.xpack.esql.inference.InputTextReader;
+import org.elasticsearch.xpack.esql.inference.bulk.BulkInferenceRequestIterator;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * This iterator reads text inputs from a {@link BytesRefBlock} and converts them into individual {@link InferenceAction.Request} instances
+ * of type {@link TaskType#TEXT_EMBEDDING}.
+ */
+public class TextEmbeddingOperatorRequestIterator implements BulkInferenceRequestIterator {
```

Comment:
These classes may be included in their Operator class as inner classes - I don't think we're going to use them outside of the operators, right?

Reply:
I did not want the main classes to become to big.
I made them package-private instead of nesting them.